src/bindgen/language_backend/clike.rs

```diff
@@ -918,26 +918,59 @@ impl LanguageBackend for CLikeLanguageBackend<'_> {
+                for (i, ordered_key) in ordered_fields.iter().enumerate() {
                     if let Some(lit) = fields.get(ordered_key) {
-                        if !is_first_field {
-                            write!(out, ", ");
-                        }
-                        is_first_field = false;
+                        let condition = lit.cfg.to_condition(self.config);
                         if self.config.language == Language::Cxx {
+                            if condition.is_some() {
+                                out.new_line();
+                            }
+                            condition.write_before(self.config, out);
                             // TODO: Some C++ versions (c++20?) now support designated
                             // initializers, consider generating them.
                             write!(out, "/* .{} = */ ", ordered_key);
+                            self.write_literal(out, &lit.value);
+                            if i + 1 != ordered_fields.len() {
+                                write!(out, ", ");
+                            }
+                            condition.write_after(self.config, out);
+                            if condition.is_some() {
+                                out.new_line();
+                            }
                         } else {
+                            if i > 0 {
+                                write!(out, ", ");
+                            }
                             write!(out, ".{} = ", ordered_key);
+                            if condition.is_some() {
+                                write!(out, "__{export_name}_{ordered_key}(");
+                                self.write_literal(out, &lit.value);
+                                write!(out, ")")
+                            } else {
+                                self.write_literal(out, &lit.value);
+                            }
                         }
-                        self.write_literal(out, lit);
                     }
                 }
                 write!(out, " }}");
+
+                if self.config.language == Language::C {
+                    for ordered_key in ordered_fields.iter() {
+                        if let Some(lit) = fields.get(ordered_key) {
+                            if let Some(condition) = lit.cfg.to_condition(self.config) {
+                                out.new_line();
+                                condition.write_before(self.config, out);
+                                let define = format!("#define __{export_name}_{ordered_key}(v)");
+                                write!(out, "{define} (v)");
+                                write!(out, "\n#else\n");
+                                write!(out, "{define}");
+                                condition.write_after(self.config, out);
+                            }
+                        }
+                    }
+                }
```

Comment:
Where is the good place to put this?
From this place, it will write duplicated definitions for each literal definition

Reply:
@emilio This part is also not resolved yet